src/genotyper.cc

```diff
@@ -108,25 +108,55 @@ class FormatFieldHelper : public IFormatFieldHelper {
         return ans;
     }
 
-    Status combine_format_data(vector<T>& ans) {
-        if (field_info.default_to_zero) {
-            for (auto& v : format_v) {
-                if (v.empty()) {
-                    v.push_back(0);
+    Status get_default_value(T* val) {
+        if (field_info.default_type == DefaultValueFiller::ZERO) {
+            *val = 0;
+        } else if (field_info.default_type == DefaultValueFiller::MISSING) {
+            switch (field_info.type) {
+                case RetainedFieldType::INT:
+                    *val = bcf_int32_missing;
+                    break;
+                case RetainedFieldType::FLOAT:
+                {
+                    float tmp = 0;
+                    bcf_float_set_missing(tmp);
+                    assert(bcf_float_is_missing(tmp));
+                    *val = tmp;
+                    break;
                 }
+                default:
+                    return Status::Invalid("genotyper: encountered unknown field_info.type");
             }
+        } else {
+            return Status::Invalid("genotyper: encountered unknown default value filler type");
         }
+        return Status::OK();
+    }
+    Status combine_format_data(vector<T>& ans) {
+        Status s;
 
-        if( any_of(format_v.begin(), format_v.end(), [](vector<T> v){return v.empty();})) {
-            // At least one of the sample is missing a value in this format field.
-            // For vcf spec compliance, we drop this format field for the output row by 
-            // clearing out the format_v vector
+        int n_empty_samples = 0;
+        // Templatized default value
+        T default_value;
+        S(get_default_value(&default_value));
+
+        for (auto& v : format_v) {
+            if (v.empty()) {
+                n_empty_samples++;
+                v.push_back(default_value);
+            }
+        }
+
+        // If all samples are missing values for this format field, we should
+        // drop it instead of filling in with default value
+        if (n_empty_samples == format_v.size()) {
```

Comment:
If we're genotyping a shard of a cohort and the shard happens to be all homozygous ref (while ALT alleles exist in some other shard) will we drop SB in the shard?